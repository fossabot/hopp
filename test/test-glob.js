/**
 * @file test/test-glob.js
 * @license MIT
 * @copyright 10244872 Canada Inc.
 */

const tmp = require('tmp');
const glob = require('../packages/hopp/dist/fs/glob').default
const cache = require('../packages/hopp/dist/cache')

describe('glob', async () => {
  it('should recursively match files relative to current dir', async () => {
    // load cache from new dir, shouldn't have lockfile
    await cache.load(tmp.dirSync().name)

    // tests with no .; these snapshots are assumed to be correct
    // in later tests
    expect(await glob('*.js', `${__dirname}/fixtures/glob`, false, true)).toMatchSnapshot()
    expect(await glob('*/*.js', `${__dirname}/fixtures/glob`, false, true)).toMatchSnapshot()
    expect(await glob('**/*.js', `${__dirname}/fixtures/glob`, false, true)).toMatchSnapshot()
  })

  it('should recursively match files relative to ./', async () => {
    // load cache from new dir, shouldn't have lockfile
    await cache.load(tmp.dirSync().name)

    async function assert (pttn) {
      const relativeToDot = await glob('./' + pttn, `${__dirname}/fixtures/glob`, false, true)
      const relativeToCurr = await glob(pttn, `${__dirname}/fixtures/glob`, false, true)

      return expect(relativeToDot).toEqual(relativeToCurr)
    }

    await assert('*.js')
    await assert('*/*.js')
    await assert('**/*.js')
  })

  it('should recursively match files relative to ../', async () => {
    // load cache from new dir, shouldn't have lockfile
    await cache.load(tmp.dirSync().name)

    async function assert (pttn) {
      const relativeToDot = await glob('../' + pttn, `${__dirname}/fixtures/glob/1`, false, true)
      const relativeToCurr = await glob(pttn, `${__dirname}/fixtures/glob`, false, true)

      return expect(relativeToDot).toEqual(relativeToCurr)
    }

    await assert('*.js')
    await assert('*/*.js')
    await assert('**/*.js')
  })
})