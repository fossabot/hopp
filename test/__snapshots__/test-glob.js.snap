// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`glob should recursively match files relative to current dir 1`] = `
Array [
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/index.js",
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/test.js",
]
`;

exports[`glob should recursively match files relative to current dir 2`] = `
Array [
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/1/index.js",
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/1/test.js",
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/index.js",
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/test.js",
]
`;

exports[`glob should recursively match files relative to current dir 3`] = `
Array [
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/1/2/3/index.js",
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/1/2/3/test.js",
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/1/2/index.js",
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/1/2/test.js",
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/1/index.js",
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/1/test.js",
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/index.js",
  "/Users/karimsa/projects/hoppjs/hopp/test/fixtures/glob/test.js",
]
`;
